### PR TITLE
[#2667]fix the bug of the data-size when the first option style has 'display: none'

### LINKED ISSUE #2667: Hide elements with data-size attribute
bug with `data-size`
there is a large list of elements with a search string, when I add `data-size`, all the elements disappear, `max-width` is added to some elements in the code and everything disappears.
all the presented elements are added via js

---

### PR DESCRIPTION
**Pull request by megumiimai**
resolved: #2667

### CODE REVIEW

File: js/bootstrap-select.js [hunk 1]
```diff
@@ -1920,7 +1920,8 @@
       if (this.selectpicker.current.data.length) {
         for (var i = 0; i < this.selectpicker.current.data.length; i++) {
           var data = this.selectpicker.current.data[i];
-          if (data.type === 'option') {
+          var a = data.element.firstChild;
```
Comment (NicolasCARPi) [inline js/bootstrap-select.js]: `a` is a poor variable name choice

**Review by NicolasCARPi — CHANGES_REQUESTED** [review]
Thank you for your contribution. This change would be a good opportunity to add a test case in the `tests` folder (even if we don't currently test automatically for things yet). Basically an example showing the problem, and showing that with this PR the problem goes away!

**Comment by megumiimai** [comment]
thank you for reviewing. I added a test (febdfea)  and pasted the capture. How about that?

### problem


### fixed

**Review by NicolasCARPi — APPROVED** [review]
Thanks, looks good to me!

**Pull request merged**